jwtFile.js

```
@@ -0,0 +1,39 @@
+const {
+    open: openFile,
+    unlink: deleteFile,
+    writeFile: writeFile,
+} = require('fs/promises');
+
+const JWT_KEY_FILE_DIRECTORY = './.temp_jwtkey';
+const NO_SUCH_FILE_OR_DIRECTORY_SYSTEM_ERROR = 'ENOENT';
+const CREATE_APPEND_FAIL_IF_EXISTS_FLAGS = 'ax';
+
+async function createJWTFile({ sourceDirectory, jwtKey }) {
+    let fileHandle;
+    try {
+        fileHandle = await openFile(
+            `${sourceDirectory}/${JWT_KEY_FILE_DIRECTORY}`,
+            CREATE_APPEND_FAIL_IF_EXISTS_FLAGS
+        );
+        await writeFile(fileHandle, jwtKey);
+        await fileHandle.close();
+    } catch (error) {
+        if (error.code === NO_SUCH_FILE_OR_DIRECTORY_SYSTEM_ERROR) {
+            throw new Error(`Directory ${sourceDirectory} does not exist`);
+        } else {
+            throw error;
+        }
+    }
+}
+
+async function deleteJWTFile({ sourceDirectory }) {
+    try {
+        await deleteFile(`${sourceDirectory}/${JWT_KEY_FILE_DIRECTORY}`);
+    } catch {}
```

I understand you want to ignore on error, but this just looks bad. It's a promise though, so the exception can be handled within the promise itself, making it look a bit better:
`await deleteFile(`${sourceDirectory}/${JWT_KEY_FILE_DIRECTORY}`).catch(() => {});`

---

It's still an empty catch, and introduces another (old) manner of dealing with promises.

---

I mean, yeah, it's just another empty catch, so it doesn't matter which we pick. I can't agree it's the _old way_ of handling things - it's handling the promise rejection, so dealing with the exception at the level it was thrown instead handling the exception thrown due to unhandled rejection. But of course, that's just semantics, if you like this one more, go for it.